src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,27 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationValidationException("User cannot be null");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationValidationException("Login must contain at least 6 characters");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new
+                    RegistrationValidationException("Password must contain at least 6 characters");
+        }
+        if (user.getAge() < 18) {
+            throw new RegistrationValidationException("Age does not meet the condition");
```

Comment:
The exception message for the age condition is not informative enough. It should include details about the minimum age requirement, similar to the good example provided in the checklist.